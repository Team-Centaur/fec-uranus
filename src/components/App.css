/* ORIGINAL */
/* :root {
  --light-main: #fff;
  --light-secondary: #f0f8ff;
  --light-text: #000;
  --light-secondary-text: #333;
  --light-overlay: rgba(255, 255, 255, 0.8);
  --light-shadow: rgba(60, 64, 67, 0.3) 0rem 0.0625rem 0.125rem 0rem,
    rgba(60, 64, 67, 0.15) 0rem 0.125rem 0.375rem 0.125rem;
  --light-primary: #ff9800;
  --light-primary-hover: #e68900;
  --light-primary-active: #cc7900;
  --light-primary-text: #fff;

  --dark-main: #202124;
  --dark-secondary: #2a2a2a;
  --dark-text: #fff;
  --dark-secondary-text: #ccc;
  --dark-overlay: rgba(32, 33, 36, 0.8);
  --dark-shadow: 0 2px 7px rgba(0, 0, 0, 0.5);
  --dark-secondary-hover: #3b3b3b;
  --dark-secondary-active: #292929;
  --dark-primary: #007bff;
  --dark-primary-hover: #0069d9;
  --dark-primary-active: #0056b3;
  --dark-primary-text: #fff;
} */

:root {
  --light-main: #fdfeff;
  --light-secondary: #e4ecf2;
  --light-text: #000;
  --light-secondary-text: #333;
  --light-secondary-hover: #d8dfe6;
  --light-secondary-active: #ccd3d9;
  --light-overlay: rgba(253, 254, 255, 0.8);
  --light-shadow: rgba(60, 64, 67, 0.3) 0rem 0.0625rem 0.125rem 0rem,
    rgba(60, 64, 67, 0.15) 0rem 0.125rem 0.375rem 0.125rem;
  --light-primary: #007bff;
  --light-primary-hover: #0069d9;
  --light-primary-active: #0056b3;
  --light-primary-text: #fff;
  --light-bg-gradient: linear-gradient(
    90deg,
    hsla(206, 35%, 92%, 1) 0%,
    hsla(210, 100%, 100%, 1) 46%,
    hsla(208, 100%, 97%, 1) 100%
  );

  --dark-main: #202124;
  --dark-secondary: #2a2a2a;
  --dark-text: #fff;
  --dark-secondary-text: #ccc;
  --dark-overlay: rgba(32, 33, 36, 0.8);
  --dark-shadow: 0 2px 7px rgba(0, 0, 0, 0.5);
  --dark-secondary-hover: #3b3b3b;
  --dark-secondary-active: #292929;
  --dark-primary: #007bff;
  --dark-primary-hover: #0069d9;
  --dark-primary-active: #0056b3;
  --dark-primary-text: #fff;

  --radius-outer: 0.4375rem;
  --radius-inner: 0.1875rem;
  /* --radius-md:
  --radius-lg: */
}

* {
  box-sizing: border-box;
  padding: 0;
  margin: 0;
}

ul {
  list-style: none;
}

main.light {
  background: var(--light-main);
  color: var(--light-text);
}

main.dark {
  background-color: var(--dark-main);
  color: var(--dark-text);
}

.app-container {
  max-width: 75rem;
  margin: 0 auto;
  padding: 2.25rem 0;
  font-family: sans-serif;
}
