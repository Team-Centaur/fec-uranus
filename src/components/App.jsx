import React, { Component } from "react";
import Overview from "./Overview/Overview.jsx";
import ItemsComponent from "./ItemsComponent/ItemsComponent.jsx";
import QuesAnswer from "./QuesAnswer/QuesAnswer.jsx";
import RatingReview from "./RatingReview/RatingReview.jsx";
class App extends Component {
  render() {
    return (
      <div>
        <h1>Hello worlds!</h1>
        <Overview />
        <ItemsComponent />
        <QuesAnswer />
        <RatingReview />
      </div>
    );
  }
}

export default App;
