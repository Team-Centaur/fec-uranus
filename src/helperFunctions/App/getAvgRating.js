import { each } from 'underscore';
const getAvgRating = (ratingsObj) => {
  let totalVotes = 0;
  let totalRating = 0;

  each(ratingsObj, (votes, key) => {
    votes = Number.parseInt(votes);
    key = Number.parseInt(key);
    const keyTotal = key * votes;
    totalVotes += votes;
    totalRating += keyTotal;
  });

  return (totalRating / totalVotes).toFixed(2);
};

export default getAvgRating;
