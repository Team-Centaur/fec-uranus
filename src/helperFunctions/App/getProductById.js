import axios from 'axios';

const apiKey = process.env.REACT_APP_API_KEY;
let getProductById = async (id) => {
  try {
    const response = await axios.get(
      `https://app-hrsei-api.herokuapp.com/api/fec2/hr-rfp/products/${id}`,
      {
        headers: { Authorization: apiKey },
      }
    );
    return response.data;
  } catch (err) {
    throw err;
  }
};
export default getProductById;
