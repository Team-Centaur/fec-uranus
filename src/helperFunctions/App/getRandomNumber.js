// Gets a random number from between and including min, max

const getRandomNumber = (min, max) => {
  const newMin = Math.ceil(min);
  const newMax = Math.ceil(max);
  return Math.floor(Math.random() * (newMax - newMin) + newMin);
};

export default getRandomNumber;
