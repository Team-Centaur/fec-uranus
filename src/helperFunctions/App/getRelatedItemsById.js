import axios from 'axios';

const apiKey = process.env.REACT_APP_API_KEY;

const getRelatedItemsById = async (id) => {
  try {
    const data = await axios({
      method: 'get',
      url: `https://app-hrsei-api.herokuapp.com/api/fec2/hr-rfp/products/${id}/related`,
      headers: {
        Authorization: apiKey,
      },
    });
    const uniqueData = [...new Set(data.data)];
    return uniqueData;
  } catch (err) {
    return err;
  }
};

export default getRelatedItemsById;
