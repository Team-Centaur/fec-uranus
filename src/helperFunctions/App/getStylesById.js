import axios from "axios";
const apiKey = process.env.REACT_APP_API_KEY;

// Async function, use .then((data) => ....)
// when running this function to get the data,
// Provide id of the product you want styles of as an argument
const getStylesById = async (id) => {
  try {
    const response = await axios({
      method: "get",
      url: `https://app-hrsei-api.herokuapp.com/api/fec2/hr-rfp/products/${id}/styles`,
      headers: { Authorization: apiKey },
    });
    return response.data;
  } catch (err) {
    return err;
  }
};
export default getStylesById;
