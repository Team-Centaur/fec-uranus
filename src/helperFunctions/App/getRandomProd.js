import axios from 'axios';
import getRandomNumber from './getRandomNumber.js';

const apiKey = process.env.REACT_APP_API_KEY;

const getRandomProd = async () => {
  try {
    const response = await axios({
      method: 'GET',
      url: 'https://app-hrsei-api.herokuapp.com/api/fec2/hr-rfp/products/?count=100',
      headers: { Authorization: apiKey },
    });
    return response.data[getRandomNumber(0, 99)];
  } catch (err) {
    return err;
  }
};

export default getRandomProd;
