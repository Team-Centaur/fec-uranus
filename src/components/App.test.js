import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import App from './App';

describe('App component', () => {
  it('Should render loading... while waiting for data', () => {
    render(<App />);
    const loadingElement = screen.getByText('Loading...');
    expect(loadingElement).toBeInTheDocument();
  });
});
