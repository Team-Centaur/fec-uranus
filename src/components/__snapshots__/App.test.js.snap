// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<div
  className="app-container"
>
  <h1>
    Hello worlds!
  </h1>
  <section
    className="overview-section"
  >
    <div
      className="promotion-container"
    />
    <div
      className="product-container"
    >
      <div
        className="image-gallery-container"
      />
    </div>
    <div
      className="description-container"
    />
  </section>
  <section
    className="items-comp--section"
  >
    <p
      style={
        {
          "fontSize": "2rem",
        }
      }
    >
      Loading...
    </p>
  </section>
  <section>
    <h2>
      QUESTIONS & ANSWERS
    </h2>
    <input
      onChange={[Function]}
      placeholder="Search for a question"
      type="search"
    />
    <button
      className="questionSearchButton"
    >
      Search
    </button>
    <button
      className="moreQuestionsButton"
      hidden={false}
      onClick={[Function]}
      type="submit"
    >
      MORE QUESTIONS
       (0)
    </button>
    <button
      className="askQuestionButton"
      onClick={[Function]}
      type="submit"
    >
      ASK A QUESTION
    </button>
    <form
      className="askQuestionContainer"
      hidden={true}
      value=""
    >
      <input
        className="askQuestionNameInput"
        maxLength="60"
        onChange={[Function]}
        placeholder="Name"
        value=""
      />
      <input
        className="askQuestionEmailInput"
        maxLength="60"
        onChange={[Function]}
        placeholder="Email"
        value=""
      />
      <textarea
        cols="50"
        maxLength="1000"
        onChange={[Function]}
        placeholder="question"
        rows="5"
        type="text"
      />
      <button
        className="submitAskQuestionButton"
        onClick={[Function]}
        type="submit"
      >
        Submit
      </button>
    </form>
  </section>
  <section
    className="ratingReview"
    id="review"
  >
    <div
      className="reviewsList"
    >
      <h1>
        Reviews List! 
      </h1>
      <button
        onClick={[Function]}
        type="button"
      >
        Show more reviews
      </button>
    </div>
    <div>
      Rating Breakdown
    </div>
    <div>
      Product Breakdown
    </div>
    <div>
      <button
        onClick={[Function]}
      >
        Write a review
      </button>
    </div>
  </section>
</div>
`;
